pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -179,6 +179,10 @@ func buildSidecarVirtualHostsForVirtualService(
 		// This is a gross HACK. Fix me. Its a much bigger surgery though, due to the way
 		// the current code is written.
 		serviceByPort[80] = nil
```

Comment:
Not sure what's this for?

Reply:
I mean without this patch

Reply:
@hzxuzhonghu do you mean to say you don't understand why this isn't just removed as part of this PR?

Reply:
I mean i donot understand why this was added previously. In my understanding, wildcard virtual host is to match a specific service, if a service name already exists, why generate extra wildcarded domains.

Reply:
Please have a look at issue #14033 that triggered this patch.